pyproject.toml

```diff
@@ -249,6 +249,10 @@ filterwarnings = [
     # ignore internal nbmake warnings
     'ignore:unclosed \<socket.socket:ResourceWarning',
     'ignore:unclosed event loop \<:ResourceWarning',
+    # ignore warnings generated while running tests
+    "ignore::DeprecationWarning",
```

Comment:
We should not ignore `DeprecationWarning`s, IMO – they serve as reminders for imminent API changes from dependencies upstream. We do not want our CI to break all of a sudden

Reply:
Unresolving this conversation, @prady0t – if we don't have a lot of `DeprecationWarning`s, could you fix them here? If there are many, then feel free to take that up in a follow-up PR.

Reply:
There are 976 errors but for only one reason.  

DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html

We need to remove `pkg_resources`

Reply:
Ah, right. We can't do that at this time due to `pybtex`